zng/resolver/context.go

```diff
@@ -448,10 +470,14 @@ func (c *Context) parseRecordTypeBody(in string) (string, zng.Type, error) {
 			continue
 		}
 		rest, ok = match(rest, "]")
-		if ok {
-			return rest, c.LookupTypeRecord(columns), nil
+		if !ok {
+			return "", nil, zng.ErrTypeSyntax
 		}
-		return "", nil, zng.ErrTypeSyntax
+		typ, err := c.LookupTypeRecord(columns)
```

Comment:
Just a few lines above (can't add a comment there), I think you can remove this?

```go
for _, c := range columns {
	if col.Name == c.Name {
              return "", nil, zng.ErrDuplicateFields
        }
}
```

Reply:
and probably you can also get rid of zng.ErrDuplicateFields